docs/DeveloperGuide.adoc

```diff
@@ -153,6 +153,38 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== [Proposed] Finding Free Timeslot feature
+==== Proposed Implementation
+
+The finding free timeslot feature is implemented by `ScheduleGenerator`. It generates the current `WeekSchedule` of a group and the common free timeslot among its members and updates the `ScheduleView` to be displayed to the user. Additionally, it implements the following operations:
+
+* `ScheduleGenerator#getFreeTimeslot()` -- Generates a `weekSchedule` of a group and updates the `ScheduleView`.
+
+These operations are exposed in the `Model` interface as `Model#getFreeTimeslot()`
+
+Firstly, the `getFreeTimeslot()` method takes in as input an ArrayList of `Person` and updates the ScheduleView of the TimeBook. When the method is executed, it checks the schedule of each `Person` in the TimeBook and generates a new combined `WeekSchedule` of the whole list of `Person`. Also, only `TimeSlots` that are within the week are checked and put into the 'WeekSchedule'. The generated `WeekSchedule` will then contain the individual schedules of each `Person` in the list for the week.
+
+After the `WeekSchedule` has been generated, the method now generates a new schedule of the free timeslots within the `WeekSchedule`. It checks the `WeekSchedule` and generates a new `FreeTimeSchedule` with contains all the timeslots in which there are no clashes with the `weekSchedule`. It also only generates `FreeTimeslots` for the week.
+
+Additionally, the generated `FreeTimeSchedule` also contains the information of the last venue the `Person` has been. This information is then used to calculate the closest location for all `Person` to meet.
```

Comment:
Perhaps a brief description can be given to describe how you find the closest location for all of them to meet.